### ISSUE TITLE
Error in function call: Call to function "jsondecode" failed: EOF

### ISSUE THREAD

**Issue by DaveHewy** [terraform-linters/tflint#708]
tflint seems to be throwing errors on assumed elements of certain resources. I.E. for the aws_iam_policy resources its assuming a heredoc must be present? so it sniffs for a closing heredoc tag

I am using a module, and sneakily parsing jsonencode/jsondecode string as the policy so that I can centralise the module and benefit from some simplicity and conformity on naming/tagging of those resources. This error was un-expected.

```hcl
Error: Failed to check `aws_iam_policy_invalid_policy` rule: Failed to eval an expression in modules/aws/iam/policy/main.tf:11; Error in function call: Call to function "jsondecode" failed: EOF.
FAIL. tflint failure.

```
## Code example
```
// define this resource in a module
// feed it any ${jsonencode(data.aws_iam_policy_document.testing.json)
resource "aws_iam_policy" "default" {
  name        = "test"
  description = "test-description"
  path        = "/lambda/"
  policy      = jsondecode(var.policy)
}
```
## Version

TFLint v0.9.3
tried the above initially then upgraded to TFLint version 0.13.2 and retried. still the same.
Terraform v0.12.17

**Comment by wata727** [comment]
Umm... I tried to run against a similar code but didn't reproduce the problem.

```hcl
variable "policy" {
  default = <<JSON
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "",
            "Effect": "Allow",
            "Action": "sts:AssumeRole",
            "Resource": "*",
            "Condition": {
                "StringEquals": {"iam:ResourceTag/Project": "ExampleCorpABC"}
            }
        }
    ]
}
JSON
}

resource "aws_iam_policy" "default" {
  name        = "test"
  description = "test-description"
  path        = "/lambda/"
  policy      = jsondecode(var.policy)
}
```

An error occurs because a string is expected as the `policy` value. Probably different from the error you faced.

```
$ tflint
Failed to check `aws_iam_policy_invalid_policy` rule. An error occurred:

Error: Failed to eval an expression in example.tf:24; Incorrect value type: Invalid expression value: string required.
```

A similar error is obtained when the `var.policy` is broken JSON value:

```hcl
variable "policy" {
  default = <<JSON
{
JSON
}

resource "aws_iam_policy" "default" {
  name        = "test"
  description = "test-description"
  path        = "/lambda/"
  policy      = jsondecode(var.policy)
}
```

```
$ tflint
Failed to check `aws_iam_policy_invalid_policy` rule. An error occurred:

Error: Failed to eval an expression in example.tf:11; Error in function call: Call to function "jsondecode" failed: EOF.
```

Is the value of `var.policy` assigned properly? Is the template valid in Terraform?
If the problem cannot be resolved, please share the minimal example that can reproduce the problem. Thanks.

**Comment by DaveHewy** [comment]
Thanks for the reply. That isn't running quite the same test.

Create the dir `modules/iam_policy`
> main.tf
```
resource "aws_iam_policy" {
  name = "test"
  description = "test"
  path = "/lambda/"
  policy = jsondecode(var.policy)
}
```
> variables.tf
```
variable "policy" {
   type = string
   default = ""
}
```
Then create your caller
```
data "aws_iam_policy_document" "example" {
  statement {
    actions = [
      "ec2:DescribeVpcs"
    ]
    resources = ["*"]
  }
}

module "example" {
  src = "modules/iam_policy"
  policy = jsonencode(data.aws_iam_policy_document.example.json)
}

**Comment by wata727** [comment]
Thank you for sharing the example. Understood.

As you say, when parsing `modules/iam_policy/main.tf`, the` jsondecode` function fails because `var.policy` is an empty string. This is the intended behavior.

If the default value of `var.policy` is never used, I recommend that you omit the default value like the following:

```hcl
variable "policy" {
   type = string
   // default = ""
}
```

In this case, TFLint doesn't evaluate the `jsondecode` function because it marks the `var.policy` as an "unknown".

Does this resolve the issue? If you still have problems please let me know. Thank you.

**Comment by wata727** [comment]
Close this due to no activity.
